crates/bevy_pbr/src/render/fog.wgsl

```diff
@@ -0,0 +1,86 @@
+#define_import_path bevy_pbr::fog
+
+// Fog formulas adapted from:
+// https://learn.microsoft.com/en-us/windows/win32/direct3d9/fog-formulas
+// https://catlikecoding.com/unity/tutorials/rendering/part-14/
+
+fn linear_fog(
+    input_color: vec4<f32>,
+    distance: f32,
+    scattering: vec3<f32>,
+) -> vec4<f32> {
+    var fog_color = fog.base_color;
+    if (fog.directional_light_color.a > 0.0) {
+        fog_color = vec4<f32>(
+            fog.base_color.rgb
+                + scattering * fog.directional_light_color.rgb * fog.directional_light_color.a,
+            fog_color.a
+        );
+    }
```

Comment:
this block is repeated 4 times, can we move it up to pbr_functions (or make a function here for it)?

also i don't think we need the branch here but since it's a uniform branch maybe it's better to keep it anyway ..?

Reply:
Just extracted it, it does look much cleaner.